### PR TITLE
Add relative template path support

### PR DESCRIPTION
**Pull request by pmjones**
@harikt @koriym and others: please take a look to see if you find any problems.

@koriym especially: note the documentation change; the docs-ja folks will want to update the translation.

### CODE REVIEW

**Comment by koriym** [comment]
Update the translation with #15

File: src/TemplateCore.php [hunk 1]
```diff
@@ -175,4 +182,50 @@ protected function endSection() : void
...
+
+        array_push($this->renderStack, $name);
+        return $name;
+    }
+
+    protected function resolveDots(string $name, string $prefix) : string
+    {
+        $name = ltrim(trim($name), '/');
+
+        if (substr($name, 0, 2) === './') {
```
Comment (koriym) [inline src/TemplateCore.php]: https://www.php.net/manual/en/function.str-starts-with.php (PHP8)

```suggestion
        if (str_starts_with($name, './')) {
```

File: src/TemplateCore.php [hunk 2]
```diff
@@ -175,4 +182,50 @@ protected function endSection() : void
...
+            $name = substr($name, 2);
+            return $this->resolveDots($name, $prefix);
+        }
+
+        if ($prefix && substr($name, 0, 3) === '../') {
```
Comment (koriym) [inline src/TemplateCore.php]:
```suggestion
        if ($prefix && str_starts_with($name, '../')) {
```

File: src/TemplateCore.php [hunk 3]
```diff
@@ -175,4 +182,50 @@ protected function endSection() : void
...
+            throw new Exception\TemplateNotFound(
+                PHP_EOL . "Could not resolve dots in template name." .
+                PHP_EOL . "Original name: '{$orig}'" .
+                PHP_EOL . "Resolved into: '{$name}'" .
+                PHP_EOL . "Probably too many '../' in the original name."
+            );
+        }
+
+        array_push($this->renderStack, $name);
+        return $name;
```
Comment (koriym) [inline src/TemplateCore.php]: Avoid reassigning variables

```suggestion
        $resolvedName = $this->resolveDots($name, $prefix);

        if (strpos($resolvedName, '..') !== false) {
            throw new Exception\TemplateNotFound(
                PHP_EOL . "Could not resolve dots in template name." .
                PHP_EOL . "Original name: '{$name}'" .
                PHP_EOL . "Resolved into: '{$resolvedName}'" .
                PHP_EOL . "Probably too many '../' in the original name."
            );
        }

        array_push($this->renderStack, $resolvedName);
        return $resolvedName;
```

**Review by koriym — COMMENTED** [review]
I don't see a problem, but I made some suggestions.

**Comment by pmjones** [comment]
That was very fast @koriym thank you!

I have discovered a bug related to named collections (e.g. `email:...`) and need to delay merging until a fix is in.

**Pull request merged**